service/src/main/kotlin/io/provenance/explorer/service/TokenSupplyService.kt

```diff
@@ -1,32 +1,42 @@
 package io.provenance.explorer.service
 
 import io.provenance.explorer.domain.core.logger
-import io.provenance.explorer.domain.entities.MarkerCacheRecord
+import io.provenance.explorer.domain.entities.AccountRecord
 import io.provenance.explorer.domain.extensions.NHASH
 import io.provenance.explorer.domain.extensions.toCoinStr
 import io.provenance.explorer.domain.models.explorer.TokenSupply
 import io.provenance.explorer.grpc.v1.BankGrpcClient
-import org.jetbrains.exposed.sql.transactions.transaction
+import io.provenance.explorer.grpc.v1.MarkerGrpcClient
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
 import org.springframework.stereotype.Service
 
 @Service
-class TokenSupplyService(private val bankClient: BankGrpcClient) {
+class TokenSupplyService(private val bankClient: BankGrpcClient, private val markerClient: MarkerGrpcClient) {
 
     protected val logger = logger(TokenSupplyService::class)
 
-    fun getTokenStats() = transaction {
-        // retrieve nhash marker address
-        val markerAddress = MarkerCacheRecord.findByDenom(NHASH)?.markerAddress!!
-        val markerAmount = bankClient.getMarkerBalance(markerAddress, NHASH).toBigDecimal()
-
+    fun getTokenStats() = runBlocking {
         val currentSupply = bankClient.getCurrentSupply(NHASH).toBigDecimal()
         val communityPoolSupply = bankClient.getCommunityPoolAmount(NHASH).toBigDecimal()
         val (stakingTotal, bonded) = bankClient.getStakingPool().pool.let { pool ->
             pool.bondedTokens.toBigDecimal().plus(pool.notBondedTokens.toBigDecimal()) to
                 pool.bondedTokens.toBigDecimal().toCoinStr(NHASH)
         }
 
-        val circulation = (currentSupply - communityPoolSupply - stakingTotal - markerAmount).toCoinStr(NHASH)
+        val zeroSeqAccounts = AccountRecord.findZeroSequenceAccounts()
+
+        val zeroSeqBalance = async { markerClient.getAllMarkerHolders(NHASH) }.await().asFlow()
+            .filter { zeroSeqAccounts.contains(it.address) }
```

Comment:
Only standard accounts in here. Not modules or marker typed.

Reply:
okie dokie